src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -262,6 +263,8 @@ private function dumpValue($value)
             return $code;
         } elseif ($value instanceof Reference) {
             return $this->getServiceCall((string) $value, $value);
+        } elseif ($value instanceof EnvVariable) {
+            return sprintf('$%s$', (string) $value);
```

Comment:
The `escape` method should be updated to also escape a value starting with '$', no?